src/main/java/com/amazonaws/services/kinesis/clientlibrary/proxies/ShardClosureVerificationResponse.java

```diff
@@ -0,0 +1,27 @@
+package com.amazonaws.services.kinesis.clientlibrary.proxies;
+
+import com.amazonaws.services.kinesis.model.Shard;
+import lombok.Getter;
+
+import java.util.List;
+
+/**
+ * ShutDown task verifies if the shard in context is necessarily closed before checkpointing the shard at SHARD_END.
+ * The verification is done by IKinesisProxy and this class is used to wrap the verification results.
+ */
+@Getter
+public class ShardClosureVerificationResponse {
```

Comment:
Can we make ShardClosureVerificationResponse an interface which only has a method isShardClosed() and use this in IKinesisProxy. Any concrete impl like  KinesisProxy can choose to return an impl of ShardClosureVerificationResponse. This way we dont need to tie listofshards to the response, as we will be deprecating this in future.

Reply:
Refactored.